src/Packetizer/ThinPacketizer.c

```diff
@@ -0,0 +1,127 @@
+ * Version 2.3 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://essr.esa.int/license/list
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "Packetizer.h"
+
+#include <assert.h>
+#include <string.h>
+
+#include "PacketizerInternal.h"
+#include "SpacePacketInternal.h"
+
+#define THIN_PACKET_PRIMARY_HEADER_SIZE 4u
+
+void
+Packetizer_init(Packetizer* const self)
+{
+}
+
+size_t
+Packetizer_packetize(Packetizer* const self,
+                     const Packetizer_PacketType packetType,
+                     const uint16_t source,
+                     const uint16_t destination,
+                     uint8_t* const packetPointer,
+                     const size_t dataOffset,
+                     const size_t dataSize)
+{
+    // Unused in this implementation
+    (void)self;
+    (void)packetType;
+    (void)source;
+    (void)dataOffset;
+
+    assert(destination <= SPACE_PACKET_MAX_APID);
+    assert(packetPointer != NULL);
+    assert(dataOffset == THIN_PACKET_PRIMARY_HEADER_SIZE);
+    assert(dataSize < SPACE_PACKET_MAX_PACKET_DATA_SIZE);
+
+    memset(packetPointer, 0, THIN_PACKET_PRIMARY_HEADER_SIZE);
+
+    writePacketId(packetPointer, packetType, destination);
+    writePacketDataLength(packetPointer, dataSize);
+
+    return THIN_PACKET_PRIMARY_HEADER_SIZE + dataSize;
+}
+
+bool
+Packetizer_depacketize(const Packetizer* const self,
+                       const Packetizer_PacketType packetType,
+                       const uint8_t* const packetPointer,
+                       const size_t packetSize,
+                       uint16_t* const source,
+                       uint16_t* const destination,
+                       size_t* const dataOffset,
+                       size_t* const dataSize,
+                       int32_t* const errorCode)
+{
+    // Unused in this implementation
+    (void)self;
+    (void)packetType;
+    (void)source;
+
+    assert(packetPointer != NULL);
+    assert(dataOffset != NULL);
+    assert(dataSize != NULL);
+    assert(destination != NULL);
+
+    if(packetSize < THIN_PACKET_PRIMARY_HEADER_SIZE) {
+        // the packet is smaller than expected (at least header)
+        return false;
+    }
+
+    // Get and check data size
+    const size_t receivedDataSize = readPacketDataLength(packetPointer);
+    if(packetSize != THIN_PACKET_PRIMARY_HEADER_SIZE + receivedDataSize) {
+        if(errorCode != NULL) {
+            *errorCode = Packetizer_ErrorCode_IncorrectPacketSize;
+        }
+        return false;
+    }
+
+    // Save the results
+    *destination = (packetPointer[0] << 8u) | packetPointer[1];
```

Comment:
readPacketId for symmetry?

Reply:
Actually the asymmetry is propagated from default Packetizer. There is no such function prototype (readPacketId) in common header (PacketizerInternal.h) for some reason. So, should I leave as is, or should I add the common prototype and edit standard Packetizer as well?

Reply:
@Lurkerpas - I don't want to impose any impact on existing code, but this seem as some sort of non-consistency. Opinions?

Reply:
I agree with @hcorg . Please add the function for symmetry, but only in the ThinPacketizer.

Reply:
fixed